### ISSUE TITLE
[Enhancement] Expose match method

### ISSUE THREAD

**Issue by tobeorla** [vuejs/vue-router#678]
It would be cool to become able to use router's match method, so i can get more info of the given route.

Example:

``` js
const router = new VueRouter({
    hashbang: false,
    history: true,
    saveScrollPosition: true,
});

router.map({
        '/': {
            name: 'home',
            component: AutoLoad.Pages.Home()
        },
        '/settings': {
            name: 'settingsPage',
            component: AutoLoad.Pages.Settings()
        }
);


var route = router.match("/settings");
console.log(route); // expected to receive the same object from map declaration
```

**Comment by yyx990803** [comment]
FYI this is available in the 2.0 router

**Comment by tobeorla** [comment]
Enough for me! ty!
